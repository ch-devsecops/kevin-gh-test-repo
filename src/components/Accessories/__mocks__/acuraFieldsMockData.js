export const acuraOriginalData = {};

export const acuraMappedData = {};

export const acuraMappedConverted = {
  anchorId: '',
  gtmTitle: '',
  gtmCategory: {
    value: 'model interactions',
  },
  categories: [
    {
      name: 'Interior',
    },
    {
      name: 'Exterior',
    },
    {
      name: 'Packages',
    },
  ],
  modelYear: {
    year: '2022',
    model: {
      name: 'MDX',
    },
    defaultTrim: {
      detKey: 'mdx_10259',
      detIdentifier: '10259',
      sotId: '',
    },
    trims: [
      {
        detKey: 'mdx_10259',
        detIdentifier: '10259',
        sotId: '',
        name: 'MDX',
        trimName: 'MDX',
        bodyType: {
          item: [
            {
              name: 'SUV',
            },
          ],
        },
        defaultTransmission: {
          detKey: '10234-Automatic',
          detIdentifier: '10234',
          sotId: '',
        },
      },
      {
        detKey: 'tech_10260',
        detIdentifier: '10260',
        sotId: '',
        name: 'TECH',
        trimName: '',
        bodyType: {
          item: [
            {
              name: 'SUV',
            },
          ],
        },
        defaultTransmission: {
          detKey: '10235-Automatic',
          detIdentifier: '10235',
          sotId: '',
        },
      },
      {
        detKey: 'a_spec_10261',
        detIdentifier: '10261',
        sotId: '',
        name: 'A-SPEC',
        trimName: '',
        bodyType: {
          item: [
            {
              name: 'SUV',
            },
          ],
        },
        defaultTransmission: {
          detKey: '10236-Automatic',
          detIdentifier: '10236',
          sotId: '',
        },
      },
      {
        detKey: 'platinum_elite_10262',
        detIdentifier: '10262',
        sotId: '',
        name: 'PLATINUM ELITE',
        trimName: '',
        bodyType: {
          item: [
            {
              name: 'SUV',
            },
          ],
        },
        defaultTransmission: {
          detKey: '10237-Automatic',
          detIdentifier: '10237',
          sotId: '',
        },
      },
      {
        detKey: 'type_s_10283',
        detIdentifier: '10283',
        sotId: '',
        name: 'TYPE S',
        trimName: '',
        bodyType: {
          item: [
            {
              name: 'SUV',
            },
          ],
        },
        defaultTransmission: {
          detKey: '10258-Automatic',
          detIdentifier: '10258',
          sotId: '',
        },
      },
      {
        detKey: 'type_s_ultra_10284',
        detIdentifier: '10284',
        sotId: '',
        name: 'TYPE S ULTRA',
        trimName: '',
        bodyType: {
          item: [
            {
              name: 'SUV',
            },
          ],
        },
        defaultTransmission: {
          detKey: '10259-Automatic',
          detIdentifier: '10259',
          sotId: '',
        },
      },
    ],
  },
};
