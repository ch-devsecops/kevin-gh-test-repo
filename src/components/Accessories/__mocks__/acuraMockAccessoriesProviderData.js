export default [
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73876_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73876_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73876_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73876_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '13908_cargo_organizer',
    AccessoryCategory: {
      DisplayOrder: 108,
      Key: 'interior',
      Name: 'Interior',
      GtmName: 'Interior',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription:
      'The Cargo Organizer offers an ideal way to help keep easily spillable items upright in the back section of your MDX.\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.1,
    AccessoryId: 13908,
    AccessoryIsResidualized: false,
    AccessoryKey: 'cargo_organizer',
    AccessoryListPrice: 129.0,
    AccessoryName: 'Cargo Organizer',
    AccessoryGtmName: 'Cargo Organizer',
    AccessoryNumber: '08U20-STK-200',
    AccessoryProvincialPrice: 145.3,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 7,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73848_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73848_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73848_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73848_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '13957_crossbars_storage_bag',
    AccessoryCategory: {
      DisplayOrder: 109,
      Key: 'exterior',
      Name: 'Exterior',
      GtmName: 'Exterior',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription: 'Conveniently store your crossbars in this handy storage bag when not in use.\r\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.0,
    AccessoryId: 13957,
    AccessoryIsResidualized: false,
    AccessoryKey: 'crossbars_storage_bag',
    AccessoryListPrice: 145.0,
    AccessoryName: 'Crossbars - Storage Bag',
    AccessoryGtmName: 'Crossbars - Storage Bag',
    AccessoryNumber: '08U51-TJB-200',
    AccessoryProvincialPrice: 145.0,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 41,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73845_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73845_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73845_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73845_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '13961_roof_rack__fork_mount_bike_attachment',
    AccessoryCategory: {
      DisplayOrder: 109,
      Key: 'exterior',
      Name: 'Exterior',
      GtmName: 'Exterior',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription:
      'The Fork Mount Bike Attachment features a same side adjustment dial that provides fast loading and maximum security for your bike.  Requires Roof Rails and Crossbars.\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.4,
    AccessoryId: 13961,
    AccessoryIsResidualized: false,
    AccessoryKey: 'roof_rack__fork_mount_bike_attachment',
    AccessoryListPrice: 403.0,
    AccessoryName: 'Bike Attachment - Fork Mount',
    AccessoryGtmName: 'Bike Attachment - Fork Mount',
    AccessoryNumber: '08L07-E09-400A',
    AccessoryProvincialPrice: 468.2,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 14,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73885_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73885_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73885_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73885_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '14001_optimate_12v_auto_battery_charger_maintainer',
    AccessoryCategory: {
      DisplayOrder: 109,
      Key: 'exterior',
      Name: 'Exterior',
      GtmName: 'Exterior',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription: 'Extended battery life by charging/maintaining an Acura battery to ideal charge levels.\r\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.0,
    AccessoryId: 14001,
    AccessoryIsResidualized: false,
    AccessoryKey: 'optimate_12v_auto_battery_charger_maintainer',
    AccessoryListPrice: 133.0,
    AccessoryName: 'OPTIMATE 12V AUTO BATTERY CHARGER/MAINTAINER',
    AccessoryGtmName: 'OPTIMATE 12V AUTO BATTERY CHARGER/MAINTAINER',
    AccessoryNumber: '31670-BMS-004A',
    AccessoryProvincialPrice: 133.0,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 50,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73838_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73838_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73838_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73838_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '14119_cold_protection_package',
    AccessoryCategory: {
      DisplayOrder: 105,
      Key: 'packages',
      Name: 'Packages',
      GtmName: 'Packages',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription: ' Package includes: Engine Block Heater and Optimate 12V Auto Battery Charger/Maintainer.\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 2.0,
    AccessoryId: 14119,
    AccessoryIsResidualized: true,
    AccessoryKey: 'cold_protection_package',
    AccessoryListPrice: 283.0,
    AccessoryName: 'COLD WEATHER PACKAGE - 3.5L',
    AccessoryGtmName: 'COLD WEATHER PACKAGE - 3.5L',
    AccessoryNumber: '22MDX-CLD-PKG',
    AccessoryProvincialPrice: 609.0,
    AccessoryResidualAmount: 500.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 3,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73839_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73839_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73839_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73839_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '14120_launch_package_a___mdx',
    AccessoryCategory: {
      DisplayOrder: 105,
      Key: 'packages',
      Name: 'Packages',
      GtmName: 'Packages',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription:
      'Package includes: Cargo Cover, Illuminated Door Sill Trims (ACURA), and Running Boards with Attachments in Chrome.',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 3.0,
    AccessoryId: 14120,
    AccessoryIsResidualized: false,
    AccessoryKey: 'launch_package_a___mdx',
    AccessoryListPrice: 2362.0,
    AccessoryName: 'Enhanced Package - MDX',
    AccessoryGtmName: 'Enhanced Package - MDX',
    AccessoryNumber: '22MDX-ENH-PKG1',
    AccessoryProvincialPrice: 2851.0,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 6,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73840_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73840_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73840_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73840_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '14123_protection_package_a___mdx',
    AccessoryCategory: {
      DisplayOrder: 105,
      Key: 'packages',
      Name: 'Packages',
      GtmName: 'Packages',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription: ' Package includes: 1st, 2nd, and 3rd Row All-Season Floor Mats and Folding Cargo Tray.',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.3,
    AccessoryId: 14123,
    AccessoryIsResidualized: false,
    AccessoryKey: 'protection_package_a___mdx',
    AccessoryListPrice: 659.0,
    AccessoryName: 'Protection Package - MDX (Out of Stock)',
    AccessoryGtmName: 'Protection Package - MDX (Out of Stock)',
    AccessoryNumber: '22MDX-PRO-PKG1',
    AccessoryProvincialPrice: 707.9,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 9,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73872_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73872_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73872_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73872_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '14154_all_season_floor_mat__high_wall___mdx',
    AccessoryCategory: {
      DisplayOrder: 108,
      Key: 'interior',
      Name: 'Interior',
      GtmName: 'Interior',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription:
      ' An ideal way to save wear and tear on your original carpets, the All-Season Floor Mats are designed to trap and hold dirt, sand, mud and water.\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.1,
    AccessoryId: 14154,
    AccessoryIsResidualized: false,
    AccessoryKey: 'all_season_floor_mat__high_wall___mdx',
    AccessoryListPrice: 309.0,
    AccessoryName: 'All-Season Floor Mats, MDX - 1st & 2nd Row (Out of Stock)',
    AccessoryGtmName: 'All-Season Floor Mats, MDX - 1st & 2nd Row (Out of Stock)',
    AccessoryNumber: '08P17-TYA-210',
    AccessoryProvincialPrice: 325.3,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 1,
  },
  {
    TransmissionId: 10280,
    OverlayAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73873_ovr_off.jpg',
    ThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73873_thumbnail.jpg',
    ThumbnailColorAssetUrl: null,
    DefaultThumbnailAssetUrl: 'https://images.honda.ca:443/accessories/A/en_Default_thumbnail.jpg',
    FrontOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73873_ovr_front_on.png',
    FrontOnColorAssetUrl: null,
    BackOnAssetUrl: 'https://images.honda.ca:443/accessories/A/2024/73873_ovr_back_on.png',
    BackOnColorAssetUrl: null,
    AccessoryBusinessKey: '14156_all_season_floor_mat__high_wall___3rd_row',
    AccessoryCategory: {
      DisplayOrder: 108,
      Key: 'interior',
      Name: 'Interior',
      GtmName: 'Interior',
    },
    AccessoryDefaultTireType: 'N',
    AccessoryDescription:
      ' An ideal way to save wear and tear on your original carpets, the All-Season Floor Mats are designed to trap and hold dirt, sand, mud and water.\n',
    AccessoryDiscountPercent: 0.0,
    AccessoryFrt: 0.1,
    AccessoryId: 14156,
    AccessoryIsResidualized: false,
    AccessoryKey: 'all_season_floor_mat__high_wall___3rd_row',
    AccessoryListPrice: 122.0,
    AccessoryName: 'All-Season Floor Mat - 3rd Row',
    AccessoryGtmName: 'All-Season Floor Mat - 3rd Row',
    AccessoryNumber: '08P17-TYA-210A',
    AccessoryProvincialPrice: 138.3,
    AccessoryResidualAmount: 0.0,
    Dependencies: [],
    Exclusions: [],
    AccessoryDisplayOrder: 3,
  },
];
