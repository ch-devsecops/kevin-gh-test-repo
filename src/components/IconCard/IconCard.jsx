import React from 'react';
import PropTypes from 'prop-types';
import { Box, Markdown, MarkdownHeading, Optional } from '@honda-canada/design-system-react';
import themeStyles from './IconCard.styles';
import { getGtmTagValue } from '../../utils/gtmEvents';

const Container = themeStyles.apply(Box, 'Container');
const Wrapper = themeStyles.apply(Box, 'Wrapper');
const IconContainer = themeStyles.apply(Box, 'IconContainer');
const BodyTextContainer = themeStyles.apply(Box, 'BodyTextContainer');
const CTAGroup = themeStyles.apply(Box, 'CTAGroup');
const Divider = themeStyles.apply(Box, 'Divider');
const Title = themeStyles.apply(MarkdownHeading, 'Title');

const IconCard = ({ icon, title, bodyText, ctas, hasDivider, backgroundColor, gtmTags }) => {
  const hasCTA = !!ctas.length;
  const hasIcon = !!icon;

  return (
    <Wrapper
      hasIcon={hasIcon}
      data-gtm-component-type={getGtmTagValue(gtmTags?.type)}
      data-gtm-category={getGtmTagValue(gtmTags?.category)}
    >
      <IconContainer>{icon}</IconContainer>

      <Container bg={backgroundColor} hasCTA={hasCTA} hasIcon={hasIcon}>
        <Box flexGrow="1">
          <Title mb={hasDivider ? 0 : ['xl', 'l']} headingOverride="h5">
            {title}
          </Title>

          <Optional when={hasDivider}>
            <Divider />
          </Optional>

          <Optional when={bodyText.length}>
            <BodyTextContainer hasCTA={hasCTA}>
              <Markdown style={{ textAlign: 'center' }} mb={0}>
                {bodyText}
              </Markdown>
            </BodyTextContainer>
          </Optional>
        </Box>

        <CTAGroup>{ctas}</CTAGroup>
      </Container>
    </Wrapper>
  );
};

IconCard.defaultProps = {
  bodyText: '',
  ctas: [],
  gtmTags: {},
  hasDivider: false,
  backgroundColor: 'white',
};

IconCard.propTypes = {
  icon: PropTypes.element,
  title: PropTypes.string,
  /**
   * Markdown enabled text
   */
  bodyText: PropTypes.string,
  hasDivider: PropTypes.bool,
  backgroundColor: PropTypes.oneOf(['white', 'grey.5']),
  ctas: PropTypes.arrayOf(PropTypes.element),
  gtmTags: PropTypes.shape({
    category: PropTypes.string,
    type: PropTypes.string,
  }),
};

export default IconCard;
