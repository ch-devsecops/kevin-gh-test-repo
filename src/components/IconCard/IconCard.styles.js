import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Wrapper')(({ hasIcon }) => {
  const styles = {
    width: '100%',
    height: '100%',
    position: 'relative',
  };

  if (hasIcon) {
    styles.pt = ['17.5px', '22.5px'];
  }

  return css(styles);
});

themeStyles.addBaseStyles('Container')(({ hasIcon, hasCTA }) => {
  const styles = {
    width: '100%',
    height: '100%',
    overflow: 'auto',
    display: 'flex',
    flexDirection: 'column',
  };

  if (!hasIcon) {
    styles.pt = ['l', '40px'];
    styles.pb = ['l', '40px'];
  } else {
    styles.pt = hasCTA ? ['40px', '55px'] : ['42px', '40px'];
    styles.pb = ['l', 'l'];
  }

  return css(styles);
});

themeStyles.addBaseStyles('Title')(
  css({
    textAlign: 'center',
    mx: 'm',
  }),
);

themeStyles.addBaseStyles('IconContainer')(
  css({
    size: ['35px', '45px'],
    position: 'absolute',
    top: 0,
    left: '50%',
    transform: 'translate(-50%, 0)',
  }),
);

themeStyles.addBaseStyles('CTAGroup')(
  css({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',

    '& > *': {
      mb: 'm',
    },

    '& > *:last-child': {
      mb: '0',
    },

    '& a': {
      color: 'red',
    },
  }),
);

themeStyles.addBaseStyles('Divider')(
  css({
    maxWidth: ['164px', '182px'],
    width: '100%',
    height: '1px',
    bg: 'primary',
    mx: 'auto',
    my: 'default',
  }),
);

themeStyles.addBaseStyles('BodyTextContainer')(({ hasCTA }) =>
  css({
    px: ['default', '14.25%'],
    mb: hasCTA ? ['m', 'default'] : [0, 0],
  }),
);

export default themeStyles;
