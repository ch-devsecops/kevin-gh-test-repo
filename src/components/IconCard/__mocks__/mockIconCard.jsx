/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/jsx-key */
import React from 'react';
import { Link, Image, Icon } from '@honda-canada/design-system-react';

export const ImageMock = () => <Image src="/images/cms-icons/PDF.png" alt="pdf-icon" />;

export const CTAMock = ({ text }) => (
  <Link styling="primary" data-gtm-title="iconcard-title">
    {text}

    <Icon ml="xs" name="save" iconColor="primary" />
  </Link>
);

export const CTAMocks = [<CTAMock text="Download English" />, <CTAMock text="Download French" />];
