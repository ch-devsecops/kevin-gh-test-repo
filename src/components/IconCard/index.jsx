import React from 'react';
import PropTypes from 'prop-types';
import { Image } from '@honda-canada/design-system-react';
import { JSSFieldPropType } from '../../utils/propTypes';
import IconCardReact from './IconCard';
import { mapGTMCategory, mapJssFieldsToCtaComponents } from '../../utils/sitecoreFields';

const IconCard = ({ fields, params, rendering }) => {
  if (!fields) {
    return null;
  }

  const { gtmTitle, gtmCategory, title, bodyText, icon } = fields;

  const gtmTags = {
    category: mapGTMCategory(gtmCategory),
    title: gtmTitle?.value,
    type: rendering?.componentName,
  };
  const ctas = mapJssFieldsToCtaComponents(fields, gtmTitle?.value, rendering?.componentName);

  return (
    <IconCardReact
      title={title?.value}
      bodyText={bodyText?.value}
      gtmTags={gtmTags}
      backgroundColor={params?.contentBgColour === 'Grey' ? 'grey.5' : 'white'}
      hasDivider={params?.hasDivider === 1}
      icon={icon.value.src && <Image {...icon.value} />}
      ctas={ctas}
    />
  );
};

IconCard.propTypes = {
  fields: PropTypes.shape({
    title: JSSFieldPropType,
    bodyText: JSSFieldPropType,
    gtmTitle: JSSFieldPropType,
    gtmCategory: JSSFieldPropType,
    icon: PropTypes.shape({
      value: PropTypes.shape({
        src: PropTypes.string,
        alt: PropTypes.string,
        width: PropTypes.string,
        height: PropTypes.string,
      }),
    }),
  }),
  params: PropTypes.shape({
    contentBgColour: PropTypes.string,
    hasDivider: PropTypes.number,
  }),
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};

export default IconCard;
