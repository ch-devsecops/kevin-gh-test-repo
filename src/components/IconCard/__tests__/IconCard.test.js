import React from 'react';
import { axe, cleanup, render, MockRendering, MockFields } from '../../../test-utils';
import { ImageMock, CTAMocks } from '../__mocks__/mockIconCard';
import IconCard from '../IconCard';

afterEach(cleanup);

const mockFields = new MockFields();
const mockRendering = new MockRendering('DiagonalCard');

describe('<IconCard>', () => {
  it('should render correctly with all props', async () => {
    const { container } = render(
      <IconCard fields={mockFields} rendering={mockRendering} title="My title" bodyText="My body text" />,
    );

    expect(await axe(container)).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });

  it('It renders body text without Icon', async () => {
    const { container } = render(
      <IconCard
        title="Quick Start Guide"
        backgroundColor="grey.5"
        hasDivider
        bodyText="Stop and exchange information."
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('It renders body text without CTA', async () => {
    const { container } = render(
      <IconCard
        title="Quick Start Guide"
        backgroundColor="grey.5"
        bodyText="Stop and exchange information."
        icon={<ImageMock />}
        hasDivider
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('It renders CTAs without body text', async () => {
    const { container } = render(
      <IconCard title="Quick Start Guide" backgroundColor="grey.5" icon={<ImageMock />} ctas={CTAMocks} />,
    );

    expect(container).toMatchSnapshot();
  });
});
