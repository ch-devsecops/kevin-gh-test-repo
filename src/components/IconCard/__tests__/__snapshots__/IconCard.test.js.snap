// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconCard> It renders CTAs without body text 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 ewpOsu"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jmUdzb"
    >
      <img
        alt="pdf-icon"
        class="sc-1k44dfv-0 jLTFZk"
        draggable="true"
        loading="lazy"
        src="/images/cms-icons/PDF.png"
      />
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 cZEkwa"
    >
      <div
        class="Box-sc-1bsf784-0 gBdgWj"
      >
        <div
          class="sc-1k44dfv-0 iBztxY sc-1k44dfv-0 dHPSTR"
          mb="xl,l"
        >
          

          <h5
            class="sc-1k44dfv-0 gXWcpP"
          >
            Quick Start Guide
          </h5>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gDtFzE"
      >
        <a
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 gPBfO Link-sc-16wd4a0-0 eYsnbh"
          data-gtm-title="iconcard-title"
        >
          Download English
        </a>
        <a
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 gPBfO Link-sc-16wd4a0-0 eYsnbh"
          data-gtm-title="iconcard-title"
        >
          Download French
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<IconCard> It renders body text without CTA 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 ewpOsu"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jmUdzb"
    >
      <img
        alt="pdf-icon"
        class="sc-1k44dfv-0 jLTFZk"
        draggable="true"
        loading="lazy"
        src="/images/cms-icons/PDF.png"
      />
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 kQbQDO"
    >
      <div
        class="Box-sc-1bsf784-0 gBdgWj"
      >
        <div
          class="sc-1k44dfv-0 gClLHi sc-1k44dfv-0 dHPSTR"
          mb="0"
        >
          

          <h5
            class="sc-1k44dfv-0 gXWcpP"
          >
            Quick Start Guide
          </h5>
        </div>
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 fluhpL"
        />
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 bXdSPe"
        >
          <div
            class="sc-1k44dfv-0 gClLHi"
            mb="0"
            style="text-align: center;"
          >
            

            <p>
              Stop and exchange information.
            </p>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gDtFzE"
      />
    </div>
  </div>
</div>
`;

exports[`<IconCard> It renders body text without Icon 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 dCCzpp"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jmUdzb"
    />
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 cSNiFm"
    >
      <div
        class="Box-sc-1bsf784-0 gBdgWj"
      >
        <div
          class="sc-1k44dfv-0 gClLHi sc-1k44dfv-0 dHPSTR"
          mb="0"
        >
          

          <h5
            class="sc-1k44dfv-0 gXWcpP"
          >
            Quick Start Guide
          </h5>
        </div>
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 fluhpL"
        />
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 bXdSPe"
        >
          <div
            class="sc-1k44dfv-0 gClLHi"
            mb="0"
            style="text-align: center;"
          >
            

            <p>
              Stop and exchange information.
            </p>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gDtFzE"
      />
    </div>
  </div>
</div>
`;

exports[`<IconCard> should render correctly with all props 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 dCCzpp"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jmUdzb"
    />
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 fRiBqD"
    >
      <div
        class="Box-sc-1bsf784-0 gBdgWj"
      >
        <div
          class="sc-1k44dfv-0 iBztxY sc-1k44dfv-0 dHPSTR"
          mb="xl,l"
        >
          

          <h5
            class="sc-1k44dfv-0 gXWcpP"
          >
            My title
          </h5>
        </div>
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 bXdSPe"
        >
          <div
            class="sc-1k44dfv-0 gClLHi"
            mb="0"
            style="text-align: center;"
          >
            

            <p>
              My body text
            </p>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gDtFzE"
      />
    </div>
  </div>
</div>
`;
